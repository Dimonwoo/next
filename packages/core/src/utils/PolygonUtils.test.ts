import { PolygonUtils } from './PolygonUtils'

describe('When finding the offset polygon', () => {
  test('When there is one point', () => {
    expect(PolygonUtils.getOffsetPolygon([[0, 0]], 10)).toStrictEqual([
      [-10, -10],
      [10, -10],
      [10, 10],
      [-10, 10],
    ])
  })

  test('When there are two points', () => {
    expect(
      PolygonUtils.getOffsetPolygon(
        [
          [0, 0],
          [10, 10],
        ],
        10
      )
    ).toMatchSnapshot()
  })

  test('When there are three points', () => {
    expect(
      PolygonUtils.getOffsetPolygon(
        [
          [0, 0],
          [10, 10],
          [-10, 10],
        ],
        10
      )
    ).toMatchSnapshot()
  })
})
