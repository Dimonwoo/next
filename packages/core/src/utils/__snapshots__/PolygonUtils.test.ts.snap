// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When finding the offset polygon When there are three points 1`] = `
Array [
  Array [
    34.14213562373095,
    20,
  ],
  Array [
    -34.14213562373095,
    20,
  ],
  Array [
    -0,
    -14.14213562373095,
  ],
]
`;

exports[`When finding the offset polygon When there are two points 1`] = `
Array [
  Array [
    7.071067811865475,
    -7.071067811865475,
  ],
  Array [
    17.071067811865476,
    2.9289321881345254,
  ],
  Array [
    2.9289321881345254,
    17.071067811865476,
  ],
  Array [
    -7.071067811865475,
    7.071067811865475,
  ],
]
`;
